src/OAuth2/Metadata/JwkSet.php

```diff
@@ -0,0 +1,96 @@
+<?php declare(strict_types=1);
+
+namespace OpenIDConnect\OAuth2\Metadata;
+
+use InvalidArgumentException;
+use JsonSerializable;
+use OpenIDConnect\OAuth2\Exceptions\OAuth2ServerException;
+
+/**
+ * JWK Set
+ *
+ * @see https://tools.ietf.org/html/rfc7517#section-5
+ */
+class JwkSet implements JsonSerializable
+{
+    /**
+     * @var array
+     */
+    private $keys = [];
+
+    /**
+     * @param array $jwkSet
+     */
+    public function __construct(array $jwkSet = [])
+    {
+        $this->init($jwkSet);
+    }
+
+    /**
+     * @param array $jwkSet
+     * @return JwkSet
+     */
+    public function init(array $jwkSet): JwkSet
+    {
+        if (isset($jwkSet['keys'])) {
+            foreach ($jwkSet['keys'] as $key) {
+                $this->add($key);
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param array $jwk
+     * @return JwkSet
+     */
+    public function add(array $jwk): JwkSet
+    {
+        if (isset($jwk['kid'])) {
+            $this->keys[$jwk['kid']] = $jwk;
+        } else {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method add uses an else expression. Else is never necessary and you can simplify the code to work without else.](https://app.codacy.com/app/oidcphp/oauth2/pullRequest?prid=4590171)